pcdsdevices/interface.py

```diff
@@ -1277,14 +1277,22 @@ def wm_pre(self):
 
             Returns
             -------
-            offset : float
+            offset : Optional[float, str]
                 How far we are from the preset position. If this is near zero,
                 we are at the position. If this positive, the preset position
                 is in the positive direction from us.
+                If the current position is unknown, return "Unknown".
             """
+            preset_pos = self.presets._cache[preset_type][name]['value']
+            # here we expect self: FltMvInterface
+            curr_pos = self.wm()
+            try:
+                pos = preset_pos - curr_pos
+            except Exception:
+                # current position is not known or unset
+                pos = "Unknown"
```

Comment:
This approach is definitely better than the status quo, but I have to wonder if a cleaner approach would have been to make a custom error message.

It's not clear to me.

At any rate, these are intended more for interactive behavior at beamlines, so eschewing an exception for a str output will often be appreciated.

Reply:
That was my reasoning here as well.  Thinking about the actual issue that spawned this bug report (overloaded callback queue delaying position update), an uninitialized position could be caused by a multitude of causes.  I think it might be hard to provide anything more useful than the "position is unknown" in an error message